godtools/App/Share/Domain/UseCases/IncrementUserCounterUseCase/IncrementUserCounterUseCase.swift

```diff
@@ -41,9 +41,19 @@ class IncrementUserCounterUseCase {
                 .eraseToAnyPublisher()
         }
         
+        // Only need to count language usage once per language.
+        if case .languageUsed = interaction, let languageAlreadyUsedCounter = userCountersRepository.getUserCounter(id: userCounterId) {
```

Comment:
Hey @rachaelblue is this essentially capping a language usage at count of 1?

Reply:
Hey @levieggertcru, yes-- this is a business rule that we only need to log language usage one time (per language), so count will be limited to 1 here.  Although per @frett, it won't hurt if we're continuing to count usage beyond 1, it's just not necessary.

Reply:
Hey @rachaelblue it doesn't hurt to cap it on our end as well.  I would probably just let the counter increment and instead on the fetch if the count is greater than 1 cap it to 1.  That way the data layer stays unchanged, we're just capping the returned value as part of the business rules.  But either way has advantages so I'm good with capping here too.

Reply:
capping the language usage counter is a really soft business rule at this point. When consuming the language counters for `UserActivity` we just check to see if a language counter is `>= 1`. I capped counting within a single tool session so that if the end user switches back and forth between parallel languages it doesn't artificially inflate the numbers.

Long term we might use these counts to determine a user's most frequently used language or something like that, at which point we might get a bit more strict on what constitutes incrementing a count, and would make capping the numbers when synced from the API untenable. But that isn't on any immediate roadmaps

Reply:
which what I described with how I count languages on Android is different than how Rachael is counting here. Reopening a tool will still increment the language counter even if it was already incremented at least once. It's just within a single tool usage session we only increment it once per language

Reply:
Hey @frett that makes sense.  How do you define a session or new session in Android?  Would changing user accounts start a new session?  Would the app have to be in the background for certain time period to start a new session?

Reply:
Hey @frett does incrementing ```.languageUsed``` include ```lessons``` and ```cyoa``` or should it be ```tracts``` only?

Reply:
I'm incrementing it for cyoa and lessons

Reply:
Okay thanks then we are good since the base ```MobileContentPagesViewModel.swift``` covers all of those.

Reply:
I did the same thing, I put it in my base renderer UI that is shared across all the type :)